lib/domain/provider/pollution/pollution_provider_model_state.dart

```diff
@@ -0,0 +1,20 @@
+import 'package:smoge/data/serialization/pollution_data.dart';
+import 'package:smoge/data/serialization/pollution_quality_index.dart';
+import 'package:smoge/data/serialization/pollution_sensor.dart';
+import 'package:smoge/data/serialization/pollution_station.dart';
+import 'package:smoge/domain/provider/provider_model_async_result.dart';
+
+class PollutionProviderModelState {
+  ProviderModelAsyncResult<PollutionStation, Exception> firstStation =
+      ProviderModelAsyncResult<PollutionStation, Exception>();
+  ProviderModelAsyncResult<List<PollutionStation>, Exception>
+      allPollutionStations =
+      ProviderModelAsyncResult<List<PollutionStation>, Exception>();
+  ProviderModelAsyncResult<List<PollutionSensor>, Exception> sensors =
+      ProviderModelAsyncResult<List<PollutionSensor>, Exception>();
+  ProviderModelAsyncResult<PollutionData, Exception> pollutionData =
+      ProviderModelAsyncResult<PollutionData, Exception>();
+  ProviderModelAsyncResult<PollutionQualityIndex, Exception>
+      pollutionQualityIndex =
+      ProviderModelAsyncResult<PollutionQualityIndex, Exception>();
```

Comment:
Ten Exception byłby nawet spoko, ale zastanawiam się czy nie fajniej byłoby zastosować podejścia z niewyciekaniem wyjątków do warstwy UI. Gadałem kiedyś na ten temat z @MiSikora chyba

Nie forsuje tego pomysłu, ale gdybym ja to pisał to wolałbym żeby to UI dostawał wiadomość np. isPollutionFetchingErrorVisible, niż sam handlował wyjątki. Tym sposobem nie wyciekamy tej części domeny do UI która nie jest w nim niezbędna, a obsługa błędów jest w warstwie logiki.

Reply:
W in10 mamy cuś takiego jak WhenPlaceError, zawiera on przetłumaczony tekst błędu, czy o to Ci chodzi?

Reply:
Dodałem SmogeError jak co